pkg/kn/commands/trigger/create.go

```diff
@@ -73,19 +72,19 @@ func NewTriggerCreateCommand(p *commands.KnParams) *cobra.Command {
 					"cannot create trigger '%s' "+
 						"because %s", name, err)
 			}
-			if filters == nil {
-				return fmt.Errorf(
-					"cannot create trigger '%s' "+
-						"because filters are required", name)
-			}
 
-			trigger := constructTrigger(name, namespace, triggerUpdateFlags.Broker, filters)
-			trigger.Spec.Subscriber = &duckv1.Destination{
+			trigger := client_v1alpha1.NewTriggerBuilder(name)
+			trigger.Namespace(namespace)
+			trigger.Broker(triggerUpdateFlags.Broker)
+			for k, v := range filters {
```

Comment:
We could conside to add a `Filters(map[string]string)` builder method so you could easily do the full chaining (and providers a nicer API for the builder)

Reply:
fixed.